grommet/grommet
# Accessible Select - DRAFT

#### What does this PR do?
Currently Select is built with a DropButton and a TextInput. Although the TextInput is readonly, this still causes an accessibility issue. Button cannot have any focusable descendants.

This PR adds an option for an accessible Select that is built using Text instead of TextInput. This PR is still a work in progress. Some of the things I am still working on:
- The original Select has a default width. I can't figure out where this is coming from. TextInput doesn't normally have a set width and when using Text the width of the Select stretches/shrinks to fit the text unless wrapped in a box with a set width.
- When a long option is selected in the original Select you can scroll to see the rest of the text. In the Select with Text if Text is wrapped with a box with a set width, you can scroll but a scroll bar shows up whereas in the original select no scroll bar shows.

#### Where should the reviewer start?
Select.js

#### What testing has been done on this PR?
Storybook
`yarn test`

#### How should this be manually tested?
Storybook Select/Simple story
Try wrapping <SelectText> with <Box width="small"> (Lines 374-382 in Select.js) to see scroll bar in Select with long option
`yarn test` after upgrading to `"jest-axe": "^5.0.1"` (Original select will fail accessibility test. Accessible Select will pass)

#### Any background context you want to provide?

#### What are the relevant issues?
Closes #5501 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
Yes

#### Should this PR be mentioned in the release notes?
Yes

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible
